### PR TITLE
network: add missing entries in routing_policy_rule_{hash,compare}_func()

### PR DESCRIPTION
**Pull request by yuwata**
This also makes `routing_policy_rule_get()` or friends take a `RoutingPolicyRule` object as an input.

### CODE REVIEW

File: src/network/networkd-manager.c [hunk 1]
```diff
@@ -851,37 +873,51 @@ int manager_rtnl_process_rule(sd_netlink *rtnl, sd_netlink_message *message, voi
                 log_warning_errno(r, "rtnl: could not get FRA_IIFNAME attribute, ignoring: %m");
                 return 0;
         }
+        if (iif) {
+                tmp->iif = strdup(iif);
+                if (!tmp->iif) {
+                        log_oom();
+                        return 0;
```
Comment (poettering) [inline src/network/networkd-manager.c]: return 0?

File: src/network/networkd-manager.c [hunk 2]
```diff
@@ -851,37 +873,51 @@ int manager_rtnl_process_rule(sd_netlink *rtnl, sd_netlink_message *message, voi
                 log_warning_errno(r, "rtnl: could not get FRA_IIFNAME attribute, ignoring: %m");
                 return 0;
         }
+        if (iif) {
```
Comment (poettering) [inline src/network/networkd-manager.c]: shorter:

```c
r = free_and_strdup(&tmp->iif, iif);
if (r < 0)
        return r;
```

File: src/network/networkd-manager.c [hunk 3]
```diff
@@ -851,37 +873,51 @@ int manager_rtnl_process_rule(sd_netlink *rtnl, sd_netlink_message *message, voi
...
                 log_warning_errno(r, "rtnl: could not get FRA_OIFNAME attribute, ignoring: %m");
                 return 0;
         }
+        if (oif) {
+                tmp->oif = strdup(oif);
+                if (!tmp->oif) {
+                        log_oom();
+                        return 0;
+                }
+        }
```
Comment (poettering) [inline src/network/networkd-manager.c]: dito

File: src/network/networkd-routing-policy-rule.c [hunk 4]
```diff
@@ -73,8 +75,13 @@ static void routing_policy_rule_hash_func(const RoutingPolicyRule *rule, struct
                 siphash24_compress(&rule->to, FAMILY_ADDRESS_SIZE(rule->family), state);
                 siphash24_compress(&rule->to_prefixlen, sizeof(rule->to_prefixlen), state);
 
+                i = rule->invert_rule;
+                siphash24_compress(&i, sizeof(i), state);
```
Comment (poettering) [inline src/network/networkd-routing-policy-rule.c]: why not just `siphash24_compress(&rule->invert_rule, sizeof(rule->invert_rule))` here?

Comment (yuwata) [reply src/network/networkd-routing-policy-rule.c]: Is it possible? `rule->invert_rule` is `bool`.

Comment (poettering) [reply src/network/networkd-routing-policy-rule.c]: should be fine. taking a pointer of a bool via `&` only fails if the bool is a bit field, but this one isn't.

Comment (yuwata) [reply src/network/networkd-routing-policy-rule.c]: For safety(?) I added `siphash24_compress_boolean()`.

**Comment by yuwata** [comment]
@poettering Thank you for reviewing this PR. I've force-pushed a new version. PTAL.

File: src/network/networkd-routing-policy-rule.c (hunk 4 above)
Comment (poettering) [reply src/network/networkd-routing-policy-rule.c]: makes sense!

**Pull request merged**